syntax = "proto3";

package wpi.proto;

import "geometry2d.proto";

option java_package = "edu.wpi.first.math.proto";

message ProtobufChassisSpeeds {
  double vx_mps = 1;
  double vy_mps = 2;
  double omega_rps = 3;
}

message ProtobufDifferentialDriveKinematics {
  double track_width_meters = 1;
}

message ProtobufDifferentialDriveWheelSpeeds {
  double left_mps = 1;
  double right_mps = 2;
}

message ProtobufMecanumDriveKinematics {
  ProtobufTranslation2d front_left = 1;
  ProtobufTranslation2d front_right = 2;
  ProtobufTranslation2d rear_left = 3;
  ProtobufTranslation2d rear_right = 4;
}

message ProtobufMecanumDriveMotorVoltages {
  double front_left_volts = 1;
  double front_right_volts = 2;
  double rear_left_volts = 3;
  double rear_right_volts = 4;
}

message ProtobufMecanumDriveWheelPositions {
  double front_left_meters = 1;
  double front_right_meters = 2;
  double rear_left_meters = 3;
  double rear_right_meters = 4;
}

message ProtobufMecanumDriveWheelSpeeds {
  double front_left_mps = 1;
  double front_right_mps = 2;
  double rear_left_mps = 3;
  double rear_right_mps = 4;
}

message ProtobufSwerveDriveKinematics {
  repeated ProtobufTranslation2d modules = 1;
}

message ProtobufSwerveModulePosition {
  double distance_meters = 1;
  ProtobufRotation2d angle = 2;
}

message ProtobufSwerveModuleState {
  double speed_mps = 1;
  ProtobufRotation2d angle = 2;
}
